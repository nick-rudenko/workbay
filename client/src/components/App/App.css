.app {
  margin: 0;
  padding: 0.001em;
  background-color: #f2f1f5;
  min-height: 100vh;
}
