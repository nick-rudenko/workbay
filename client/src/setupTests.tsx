// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom/extend-expect';
import React, { ComponentType } from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore } from '@reduxjs/toolkit';
import rootReducer from './store/rootReducer';

const customRender = (ui: React.ReactElement<any>, options?: any) => {
  // Wrap dispatch in a mock so it can be spied on.
  const store = createStore(rootReducer, options?.state);
  store.dispatch = jest.fn(store.dispatch);

  const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
    return <Provider store={store}>{children}</Provider>;
  };

  const returns = render(ui, {
    wrapper: AllTheProviders as ComponentType,
    ...options,
  });

  return { store, ...returns };
};

export * from '@testing-library/react';
export { customRender as render };
