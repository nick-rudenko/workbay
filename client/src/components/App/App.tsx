import React from 'react';

import './App.css';
import NotesList from '../NotesList';

const App = () => {
  return (
    <div className="app">
      <NotesList />
    </div>
  );
};

export default App;
