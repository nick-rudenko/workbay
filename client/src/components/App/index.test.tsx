import React from 'react';
import { render } from '../../setupTests';
import App from './App';

describe('App', () => {
  test('renders correctly', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
