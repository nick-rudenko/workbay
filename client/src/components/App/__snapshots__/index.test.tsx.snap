// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <div
    class="app"
  >
    <div
      class="notes-list"
    >
      <div
        class="notes-list__form"
      >
        <div
          class="create-note"
          role="form"
        >
          <div
            class="create-note__input create-note__input--body create-note__input--initial"
            contenteditable="true"
            data-placeholder="✍️ Take a note..."
            role="textbox"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
